### PR TITLE
Reddit Pixel integration

### PR DESCRIPTION
**Pull request by akashrpo**
add RedditPixel device-mode integration

## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/272)

### CODE REVIEW

File: integrations/RedditPixel/browser.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        });
+        p.callQueue = [];
+        var t = d.createElement("script");
+        (t.src = "https://www.redditstatic.com/ads/pixel.js"), (t.async = !0);
+        var s = d.getElementsByTagName("script")[0];
+        s.parentNode.insertBefore(t, s);
+      }
+    })(window, document);
+
+    rdt("init", "${this.advertiserId}");
```
Comment (arajguha) [inline integrations/RedditPixel/browser.js:28]: `"${this.advertiserId}"` String templating is dont with "`" back ticks. So, it should be
`${this.advertiserId}`

Comment (akashrpo) [reply integrations/RedditPixel/browser.js]: corrected string templating

File: integrations/RedditPixel/browser.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        });
+        p.callQueue = [];
+        var t = d.createElement("script");
+        (t.src = "https://www.redditstatic.com/ads/pixel.js"), (t.async = !0);
+        var s = d.getElementsByTagName("script")[0];
+        s.parentNode.insertBefore(t, s);
+      }
+    })(window, document);
+
+    rdt("init", `${this.advertiserId}`);
```
Comment (arajguha) [inline integrations/RedditPixel/browser.js:27]: we can simply use this.advertiserId. No need to use string template

File: integrations/RedditPixel/browser.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  }
+
+  isLoaded() {
+    logger.debug("===In isLoaded RedditPixel===");
+    return !!(window.rdt && window.rdt.advertiserID === this.advertiserId);
+  }
+
+  isReady() {
+    logger.debug("===In isReady RedditPixel===");
```
Comment (arajguha) [inline integrations/RedditPixel/browser.js:36]: we should properly check once if this is okay or simply use `!!window.rdt`

File: integrations/RedditPixel/browser.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        window.rdt("track", "Lead");
+        break;
+      case "view content":
+        window.rdt("track", "ViewContent");
+        break;
+      case "search":
+        window.rdt("track", "Search");
+        break;
+      default:
+        logger.debug("Track call not supported");
```
Comment (arajguha) [inline integrations/RedditPixel/browser.js:74]: how about adding the invalid event name in the log string as well ?

File: integrations/RedditPixel/browser.js (hunk 2 above)
Comment (akashrpo) [reply integrations/RedditPixel/browser.js]: updated with this.advertiserId without string template

File: integrations/RedditPixel/browser.js (hunk 3 above)
Comment (akashrpo) [reply integrations/RedditPixel/browser.js:45]: both the earlier checks are working now as expected.

File: integrations/RedditPixel/browser.js (hunk 4 above)
Comment (akashrpo) [reply integrations/RedditPixel/browser.js]: It has been updated.

**Pull request merged**